src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4787,22 +5214,44 @@ void makeAndAdjustReceiverSlot(BoundExpression receiver)
 
             resultType ??= node.Type?.SetUnknownNullabilityForReferenceTypes();
 
-            NullableFlowState resultState;
-            if (!wasTargetTyped)
+            TypeWithAnnotations resultTypeWithAnnotations;
+
+            if (resultType is null)
             {
-                if (resultType is null)
+                if (!wasTargetTyped)
                 {
                     // This can happen when we're inferring the return type of a lambda. For this case, we don't need to do any work,
                     // the unconverted conditional operator can't contribute info. The conversion that should be on top of this
                     // can add or remove nullability, and nested nodes aren't being publicly exposed by the semantic model.
                     Debug.Assert(node is BoundUnconvertedConditionalOperator);
```

Comment:
Do these assertions not apply in all cases?

Reply:
> Do these assertions not apply in all cases?

As you can see, this is an existing code, I didn't change this logic, just swapped the order of evaluations of conditions in the enclosing ```if``` statements. It was not my intent to move asserts or make them stronger.

Reply:
Ah, yes, I misread the existing code. Thanks.